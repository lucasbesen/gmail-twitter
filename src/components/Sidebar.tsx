import React from 'react';
import styled from 'styled-components';

import ComposeButton from './ComposeButton';
import SidebarItem from './SidebarItem';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 20%;
  background-color: #fff;
  height: 100%;
`;

const Sidebar = () => (
  <Wrapper>
    <ComposeButton />
    <SidebarItem />
  </Wrapper>
);

export default Sidebar;
