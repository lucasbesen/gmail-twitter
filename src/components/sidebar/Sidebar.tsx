import React from 'react';
import styled from 'styled-components';

import ComposeButton from '../ComposeButton';
import SidebarItem from './SidebarItem';
import { SIDEBAR_ITEM_TYPE } from './constants';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 20%;
  background-color: #fff;
  height: 100%;
`;

const Sidebar: React.FC = () => (
  <Wrapper>
    <ComposeButton />
    <SidebarItem type={SIDEBAR_ITEM_TYPE.INBOX} isActive>
      Inbox
    </SidebarItem>
    <SidebarItem type={SIDEBAR_ITEM_TYPE.STARRED}>Starred</SidebarItem>
    <SidebarItem type={SIDEBAR_ITEM_TYPE.SNOOZED}>Snoozed</SidebarItem>
  </Wrapper>
);

export default Sidebar;
