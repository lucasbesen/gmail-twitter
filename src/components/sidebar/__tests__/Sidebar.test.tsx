import React from 'react';
import renderer from 'react-test-renderer';

import Sidebar from '../Sidebar';

it('should render Sidebar without errors', () => {
  const component = renderer.create(<Sidebar />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
