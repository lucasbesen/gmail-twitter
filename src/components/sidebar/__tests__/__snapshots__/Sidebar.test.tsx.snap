// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sidebar without errors 1`] = `
<div
  className="sc-VigVT lnsbWi"
>
  <button
    className="sc-bdVaJa KESzz"
  >
    Compose
  </button>
  <div
    className="sc-gZMcBi gpQbKK"
  >
    <svg
      aria-hidden="true"
      className="sc-iwsKbI eWCEtS StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 14 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14 9l-1.13-7.14c-.08-.48-.5-.86-1-.86H2.13c-.5 0-.92.38-1 .86L0 9v5c0 .55.45 1 1 1h12c.55 0 1-.45 1-1V9zm-3.28.55l-.44.89c-.17.34-.52.56-.91.56H4.61c-.38 0-.72-.22-.89-.55l-.44-.91c-.17-.33-.52-.55-.89-.55H1l1-7h10l1 7h-1.38c-.39 0-.73.22-.91.55l.01.01z"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="sc-gqjmRU evirGQ"
    >
      Inbox
    </span>
  </div>
  <div
    className="sc-gZMcBi ffwLUT"
  >
    <svg
      aria-hidden="true"
      className="sc-htoDjs fOJkgD StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
      />
    </svg>
    <span
      className="sc-gqjmRU eXENcs"
    >
      Starred
    </span>
  </div>
  <div
    className="sc-gZMcBi ffwLUT"
  >
    <svg
      aria-hidden="true"
      className="sc-dnqmqq hUzgCs StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 0a8 8 0 100 16A8 8 0 008 0zm2.293 11.707L7 8.414V4h2v3.586l2.707 2.707-1.414 1.414z"
      />
    </svg>
    <span
      className="sc-gqjmRU eXENcs"
    >
      Snoozed
    </span>
  </div>
</div>
`;
